python/tank_vendor/shotgun_base/utils.py

```diff
@@ -18,6 +18,34 @@
 from .log import get_shotgun_base_logger
 log = get_shotgun_base_logger()
 
+
+def with_cleared_umask(func):
+    """
+    Decorator which clears the umask for a method.
+
+    The umask controls default permissions for files
```

Comment:
Well, in this case the documentation is probably good enough that you don't need to know what the parameters and return values are, especially since there is an example. Adding param and returns might be noise/make-work at this point.